engine/schema/src/main/java/com/cloud/upgrade/dao/DatabaseAccessObject.java

```diff
@@ -101,8 +105,14 @@ public boolean indexExists(Connection conn, String tableName, String indexName)
         return false;
     }
 
-    public void createIndex(Connection conn, String tableName, String columnName, String indexName) {
-        String stmt = String.format("CREATE INDEX %s on %s (%s)", indexName, tableName, columnName);
+    public void createIndex(Connection conn, String tableName, String indexName, String... columnNames) {
+        StringBuilder stmtBuilder = new StringBuilder("CREATE INDEX " + indexName + " ON " + tableName + " (");
+        for (String column : columnNames) {
+            stmtBuilder.append(column).append(",");
+        }
+        stmtBuilder.deleteCharAt(stmtBuilder.length() - 1);
```

Comment:
```suggestion
        stmtBuilder.append(StringUtils.join(columnNames, ","));
```

need to `import org.apache.commons.lang3.StringUtils;`
not tested it

Reply:
Let me do these changes.